app/models/signal_handler.rb

```diff
@@ -0,0 +1,31 @@
+class SignalHandler
+  def initialize
+    @read, @write = IO.pipe
+    Thread.new { run }
```

Comment:
The old signal handling code didn't run in a separate thread.  Is there a reason we're now spawning a thread to execute this logic?